### ISSUE TITLE
Id was not generated till the end of blob in .id.h file using ncnn2mem  

### ISSUE THREAD

**Issue by andrew-yang0722** [Tencent/ncnn#2027]
I used onnx2ncnn to make .bin and .param file.
Then, I used "ncnn2mem" to generate .id.h file.
I expected to generate all the output blob for my model.
but I found that "474" blob is the last blob in .id.h file.
it did not make the id till the end of the blobs.
What can make this problem?
I need to extract BLOB_466 and BLOB_479.

I used the command as the below.
$ ./ncnn2mem ./onnx/mymodel.param ./onnx/mymodel.bin mymodel.id.h mymodel.men.h

"mymodel.id.h"

...
const int BLOB_466 = 247;
const int LAYER_468 = 217;
const int BLOB_468 = 248;
const int LAYER_470 = 218;
const int BLOB_470 = 249;
const int LAYER_472 = 219;
const int BLOB_472 = 250;
const int LAYER_474 = 220;
const int BLOB_474 = 251;
} // namespace fd_03_param_id
#endif // NCNN_INCLUDE_GUARD_mymodel_id_h

log of exporting "mymodel.onnx"
 %454 : Tensor = onnx::Constant[value= 1 -1  4 [ CPULongType{3} ]]()
  %455 : Float(1, 3136, 4) = onnx::Reshape(%300, %454) # x.py:148:0
  %456 : Tensor = onnx::Constant[value= 1 -1  4 [ CPULongType{3} ]]()
  %457 : Float(1, 784, 4) = onnx::Reshape(%346, %456) # x.py:148:0
  %458 : Tensor = onnx::Constant[value= 1 -1  4 [ CPULongType{3} ]]()
  %459 : Float(1, 196, 4) = onnx::Reshape(%406, %458) # x.py:148:0
  %460 : Tensor = onnx::Constant[value= 1 -1  4 [ CPULongType{3} ]]()
  %461 : Float(1, 49, 4) = onnx::Reshape(%437, %460) # x.py:148:0
  %462 : Tensor = onnx::Constant[value= 1 -1  4 [ CPULongType{3} ]]()
  %463 : Float(1, 16, 4) = onnx::Reshape(%445, %462) # x.py:148:0
  %464 : Tensor = onnx::Constant[value= 1 -1  4 [ CPULongType{3} ]]()
  %465 : Float(1, 4, 4) = onnx::Reshape(%453, %464) # x.py:148:0
  %466 : Float(1, 4185, 4) = onnx::Concat[axis=1](%455, %457, %459, %461, %463, %465) # x.py:148:0
  %467 : Tensor = onnx::Constant[value= 1 -1  2 [ CPULongType{3} ]]()
  %468 : Float(1, 3136, 2) = onnx::Reshape(%299, %467) # x.py:149:0
  %469 : Tensor = onnx::Constant[value= 1 -1  2 [ CPULongType{3} ]]()
  %470 : Float(1, 784, 2) = onnx::Reshape(%345, %469) # x.py:149:0
  %471 : Tensor = onnx::Constant[value= 1 -1  2 [ CPULongType{3} ]]()
  %472 : Float(1, 196, 2) = onnx::Reshape(%405, %471) # x.py:149:0
  %473 : Tensor = onnx::Constant[value= 1 -1  2 [ CPULongType{3} ]]()
  %474 : Float(1, 49, 2) = onnx::Reshape(%436, %473) # x.py:149:0
  %475 : Tensor = onnx::Constant[value= 1 -1  2 [ CPULongType{3} ]]()
  %476 : Float(1, 16, 2) = onnx::Reshape(%444, %475) # x.py:149:0
  %477 : Tensor = onnx::Constant[value= 1 -1  2 [ CPULongType{3} ]]()
  %478 : Float(1, 4, 2) = onnx::Reshape(%452, %477) # x.py:149:0
  %479 : Float(1, 4185, 2) = onnx::Concat[axis=1](%468, %470, %472, %474, %476, %478) # x.py:149:0
  return (%466, %479)

ncnn version: 20200727

**Comment by andrew-yang0722** [comment]
I write a simple example that does not generate output id.
Is this a bug?
I wrote the code in pytorch and export it to ONNX and I used onnx2ncnn.

python code
```
import torch
import torch.nn as nn

from contextlib import redirect_stdout

class CatModel(nn.Module):
    def __init__(self):
        super(CatModel, self).__init__()

    def forward(self, x):
        x_r = x.view(1, -1, 224).contiguous()
        x_r_e = x_r * 0.5
        out = torch.cat([x_r, x_r_e], 1)
        return out

if __name__ == "__main__":
    input_sz_h = input_sz_w = 224
    onnx_file = "cat.onnx"
    net = CatModel()
    dummy = torch.randn(1, 3, input_sz_h, input_sz_w)
    onnx_log = "log.cat_onnx"
    with open(onnx_log, 'w') as f, redirect_stdout(f):
        torch.onnx.export(net, dummy, onnx_file, verbose=True,
                          input_names=["input"], output_names=["output"],
                          do_constant_folding=True)

```

onnx log
```
graph(%input : Float(1, 3, 224, 224)):
  %1 : Tensor = onnx::Constant[value=   1   -1  224 [ CPULongType{3} ]]()
  %2 : Float(1, 672, 224) = onnx::Reshape(%input, %1) # cat_onnx.py:13:0
  %3 : Float() = onnx::Constant[value={0.5}]()
  %4 : Float(1, 672, 224) = onnx::Mul(%2, %3)
  %output : Float(1, 1344, 224) = onnx::Concat[axis=1](%2, %4) # cat_onnx.py:15:0
  return (%output)
```

cat.param file
```
  1 7767517
  2 6 7
  3 Input            input                    0 1 input
  4 MemoryData       3                        0 1 3 0=1
  5 Reshape          2                        1 1 input 2 0=224 1=-1
  6 Split            splitncnn_0              1 2 2 2_splitncnn_0 2_splitncnn_1
  7 MemoryData       3                        0 1 3 0=1
  8 BinaryOp         4                        2 1 2_splitncnn_1 3 4 0=2
  9 Concat           output                   2 1 2_splitncnn_0 4 output 0=0
```

cat.id.h file
```
  1 #ifndef NCNN_INCLUDE_GUARD_cat_id_h
  2 #define NCNN_INCLUDE_GUARD_cat_id_h
  3 namespace cat_param_id {
  4 const int LAYER_input = 0;
  5 const int BLOB_input = 0;
  6 const int LAYER_3 = 1;
  7 const int BLOB_3 = 1;
  8 const int LAYER_2 = 2;
  9 const int BLOB_2 = 2;
 10 const int LAYER_splitncnn_0 = 3;
 11 const int BLOB_2_splitncnn_0 = 3;
 12 const int BLOB_2_splitncnn_1 = 4;
 13 const int LAYER_3 = 4;
 14 const int BLOB_3 = 5;
 15 const int LAYER_4 = 5;
 16 const int BLOB_4 = 6;
 17 } // namespace cat_param_id
 18 #endif // NCNN_INCLUDE_GUARD_cat_id_h
```

**Comment by nihui** [comment]
There is a duplicated MemoryData layer in your ncnn.param file

```
MemoryData       3                        0 1 3 0=1
```


follow the guide here
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
